cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/data/LocalDataLayer.java

```diff
@@ -249,6 +258,7 @@ private LocalDataLayer(@NotNull CassandraVersion version,
                            @NotNull List<SchemaFeature> requestedFeatures,
                            boolean useBufferingInputStream,
                            @Nullable String statsClass,
+                           @Nullable TimeRangeFilter sstableTimeRangeFilter,
```

Comment:
I noticed that the patch uses `List.of()` for sstableTimeRangeFilter when not defined. But the `Nullable` here implies that null value can also be used. It is inconsistent. Please pick one; otherwise, it will bite us later. 
I would suggest to change the annotation to `@NotNull` and ensure callsites do not pass null value.